### PR TITLE
Eng 2860

### PR DESCRIPTION
**Pull request by ankitintg**
- Implemented raw data endpoint functionality in SDK.
- we can access raw_data now. An example is below-
        `raw_data_table (str) = 'cycle_raw_data'`
        `timeselection (dict) = {"time_type":"relative","relative_start":1,"relative_unit":"day","ctime_tz":"America/Los_Angeles"}`
        `select (list) = ['stats__ConveyorInput__val','stats__ConveyorOutput__val']`
        `client.get_raw_data(raw_data_table, fields=select, time_selection=timeselection)`
- Integrated unit tests as well.

### CODE REVIEW

**Comment by mrnthv** [comment]
The unit tests are failing since the commit https://github.com/sightmachine/sightmachine-sdk/commit/1e992b91bef6aeaf56b5c82215f5fd1cd1182855
We are looking into fixing them as part of ticket https://sightmachine.atlassian.net/browse/ENG-3115

**Comment by mrnthv** [comment]
> The unit tests are failing since the commit [1e992b9](https://github.com/sightmachine/sightmachine-sdk/commit/1e992b91bef6aeaf56b5c82215f5fd1cd1182855) We are looking into fixing them as part of ticket https://sightmachine.atlassian.net/browse/ENG-3115

The unit tests are fixed @mklein0 . Can you review this change to enable merging.

File: tests/raw_data/test_raw_data.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import pandas as pd
+from tests.conftest import TENANT
+from smsdk.smsdk_entities.raw_data.raw_data import RawData
+
+RAW_DATA_TABLE = "cycle_raw_data"
+NUM_ROWS = 400
+URL_V1 = "/v1/datatab/raw_data"
+
+
+def test_get_utilities(get_session):
```
Comment (mgoldmansight) [inline tests/raw_data/test_raw_data.py:10]: Just note that these tests are run against the demo ENV and will need to change if the env does

File: smsdk/client.py [hunk 2]
```diff
@@ -545,3 +545,32 @@ def create_share_link(
...
+        raw_data = smsdkentities.get("raw_data")
+        base_url = get_url(
+            self.config["protocol"], self.tenant, self.config["site.domain"]
+        )
+        select = []
+        if len(fields) > 0:
+            for field in fields:
+                dict_ = {}
+                dict_["name"] = field
+                select.append(dict_)
```
Comment (mklein0) [inline smsdk/client.py:569]: Less instructions / steps.
```suggestion
        select = [
            {
                "name": field,
            }
            for field in fields
        ]
```

File: smsdk/smsdk_entities/raw_data/raw_data.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import json
+from typing import List
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
```
Comment (mklein0) [inline smsdk/smsdk_entities/raw_data/raw_data.py:8]: FYI, PY3.7 is no longer a valid release.
- https://devguide.python.org/versions/

File: tests/raw_data/test_raw_data.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    expected_list = ["get_utilities", "get_raw_data"]
+
+    assert len(all_utilites) == len(expected_list)
+    assert all([a == b for a, b in zip(all_utilites, expected_list)])
+
+
+def test_get_raw_data(get_client):
+    raw_data = get_client.get_raw_data(RAW_DATA_TABLE)
+    assert len(raw_data) == NUM_ROWS
+    assert raw_data.shape == (400, 33)
```
Comment (mklein0) [inline tests/raw_data/test_raw_data.py:25]: Something else we can verify is if there are a `timestamp` and `_id` column.  I believe those are required fields if the select is not filtered.

Comment (ankitintg) [reply tests/raw_data/test_raw_data.py:32]: Added these checks in test case.

File: smsdk/client.py (hunk 2 above)
Comment (ankitintg) [reply smsdk/client.py]: Done!

**Review by mklein0 — APPROVED** [review]

**Pull request merged**